File: app/src/main/java/com/example/pruebaandroid/LoginActivity.java
package com.example.pruebaandroid;

import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;

public class LoginActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);

        EditText inputUsuario = findViewById(R.id.editTextText2);
        EditText inputPass = findViewById(R.id.editTextTextPassword);
        Button btnIngresar = findViewById(R.id.btnIngresar);
        Button btnRegistro = findViewById(R.id.btnCrear);

        btnRegistro.setOnClickListener(v -> {
            Intent intentRegistro = new Intent(LoginActivity.this, RegistroActivity.class);
            startActivity(intentRegistro);
        });

        btnIngresar.setOnClickListener(v -> {
            String usuarioEscrito = inputUsuario.getText().toString().trim();
            String passEscrita = inputPass.getText().toString().trim();

            if (usuarioEscrito.isEmpty() || passEscrita.isEmpty()) {
                if (usuarioEscrito.isEmpty()) inputUsuario.setError("Ingrese su correo");
                if (passEscrita.isEmpty()) inputPass.setError("Ingrese su contraseña");
            } else {
                SharedPreferences preferencias = getSharedPreferences("DatosUsuario", Context.MODE_PRIVATE);
                String usuarioGuardado = preferencias.getString("Usuario", "");
                String passGuardada = preferencias.getString("Contrasena", "");

                if (usuarioEscrito.equals(usuarioGuardado) && passEscrita.equals(passGuardada)) {
                    Intent intent = new Intent(LoginActivity.this, Bienvenida.class);
                    intent.putExtra("Correo enviado", usuarioEscrito);
                    startActivity(intent);
                } else {
                    Toast.makeText(LoginActivity.this, "Datos erróneos", Toast.LENGTH_SHORT).show();
                }
            }
        });
    }
}

File: app/src/main/java/com/example/pruebaandroid/LoginActivity.kt
package com.example.pruebaandroid

import android.content.Context
import android.os.Bundle
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import android.widget.Button
import android.content.Intent
import android.widget.EditText
import android.widget.Toast
class LoginActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_login)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }
        val inputUsuarioLogin = findViewById<EditText>(R.id.editTextText2)
        val inputPassLogin = findViewById<EditText>(R.id.editTextTextPassword)
        val botonIngresar = findViewById<Button>(R.id.btnIngresar)
        val botonRegistro = findViewById<Button>(R.id.btnCrear)

        botonRegistro.setOnClickListener {
            val intentRegistro = Intent(this, RegistroActivity::class.java)
            startActivity(intentRegistro)
        }

        botonIngresar.setOnClickListener {
            val preferencias = getSharedPreferences("DatosUsuario", Context.MODE_PRIVATE)
            val usuarioGuardado = preferencias.getString("Usuario", "")
            val passGuardada = preferencias.getString("Contrasena", "")
            val usuarioEscrito = inputUsuarioLogin.text.toString()
            val passEscrita = inputPassLogin.text.toString()
            if (usuarioEscrito == usuarioGuardado && passEscrita == passGuardada) {
                val intentBienvenida = Intent(this, Bienvenida::class.java)
                startActivity(intentBienvenida)

            } else {
                Toast.makeText(this, "Datos incorrectos", Toast.LENGTH_SHORT).show()
            }
        }
    }
}